tests/src/test/scala/com/nvidia/spark/rapids/BatchWithPartitionDataSuite.scala

```diff
@@ -54,12 +50,107 @@ class BatchWithPartitionDataSuite extends SparkQueryCompareTestSuite {
     val conf = new SparkConf(false)
       .set(RapidsConf.CUDF_COLUMN_SIZE_LIMIT.key, "1000")
     withCpuSparkSession(_ => {
-      val (rowValues, rowNums, _) = generateRowData()
-      val partitionRowData = PartitionRowData.from(rowValues, rowNums)
+      val (_, partValues, partRows, _) = getSamplePartitionData
+      val partitionRowData = PartitionRowData.from(partValues, partRows)
       val resultPartitions = BatchWithPartitionDataUtils.splitPartitionDataInHalf(partitionRowData)
       val resultRowCounts = resultPartitions.flatMap(_.map(_.rowNum)).sum
       val expectedRowCounts = partitionRowData.map(_.rowNum).sum
       assert(resultRowCounts == expectedRowCounts)
     }, conf)
   }
+
+  test("test adding partition values to batch with OOM retry") {
+    val conf = new SparkConf(false)
+      .set(RapidsConf.CUDF_COLUMN_SIZE_LIMIT.key, "1000")
+    withGpuSparkSession(_ => {
+      val (partCols, partValues, partRows, partSchema) = getSamplePartitionData
+      val (valueCols, _) = getSampleValueData
+      withResource(buildBatch(valueCols)) { valueBatch =>
+        withResource(buildBatch(partCols)) { partBatch =>
+          withResource(GpuColumnVector.combineColumns(valueBatch, partBatch)) { expectedBatch =>
+            val resultBatchIter = BatchWithPartitionDataUtils.addPartitionValuesToBatch(valueBatch,
+              partRows, partValues, partSchema)
+            withResource(resultBatchIter) { _ =>
+              RmmSpark.forceRetryOOM(RmmSpark.getCurrentThreadId)
+              val rowCounts = resultBatchIter.map(_.numRows()).toSeq
+              // Assert that the final count of rows matches expected batch
+              assert(rowCounts.sum == expectedBatch.numRows())
+            }
+          }
+        }
+      }
+    }, conf)
+  }
+
+  test("test adding partition values to batch with OOM split and retry") {
+    val conf = new SparkConf(false)
+      .set(RapidsConf.CUDF_COLUMN_SIZE_LIMIT.key, "1000")
+    withGpuSparkSession(_ => {
+      val (partCols, partValues, partRows, partSchema) = getSamplePartitionData
+      val (valueCols, _) = getSampleValueData
+      withResource(buildBatch(valueCols)) { valueBatch =>
+        withResource(buildBatch(partCols)) { partBatch =>
+          withResource(GpuColumnVector.combineColumns(valueBatch, partBatch)) { expectedBatch =>
+            val resultBatchIter = BatchWithPartitionDataUtils.addPartitionValuesToBatch(valueBatch,
+              partRows, partValues, partSchema)
+            withResource(resultBatchIter) { _ =>
+              RmmSpark.forceSplitAndRetryOOM(RmmSpark.getCurrentThreadId)
+              val rowCounts = resultBatchIter.map(_.numRows()).toSeq
+              // Assert that there was a split and the final count of rows matches expected batch
```

Comment:
how is this  asserting that a split happened?

Reply:
Updated the tests to have two scenarios:
1. Catch `SplitAndRetryOOM` when partition values with single rows are passed
    => This ensures that retry is actually triggered 
3. Process the batch successfully in other cases.
    => This ensures that after retry successful processing occurs.